playground/AspireEventHub/EventHubsConsumer/Processor.cs

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Azure.Messaging.EventHubs;
+using Azure.Storage.Blobs;
+
+namespace EventHubsConsumer;
+
+/// <summary>
+///   Demonstrates how to use the <see cref="EventProcessorClient"/> to process events from an Azure Event Hub.
+/// </summary>
+/// <remarks>
+///   The EventProcessorClient in Azure Event Hubs is a powerful tool that allows for the processing of events from multiple partitions concurrently.
+///   It’s crucial to spread the reading of partitions across multiple nodes for load balancing and to maximize throughput.
+///   This is achieved by assigning each partition to a specific node, ensuring that the workload is evenly distributed.
+///   The EventProcessorClient is configured with a <see cref="BlobContainerClient"/> to store checkpoint and ownership data.
+///   Use <see cref="EventProcessorClientOptions"/> to specify options such as the maximum wait time and prefetch count.
+///   Proper configuration and usage of the EventProcessorClient can significantly improve the efficiency and performance of your Azure Event Hubs
+///   applications. Remember to handle exceptions and monitor the health of your nodes to ensure smooth operation.
+/// </remarks>
+/// <example>
+///   See samples at https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples
+/// </example>
+/// <param name="client"></param>
+/// <param name="logger"></param>
+internal sealed class Processor(EventProcessorClient client, ILogger<Consumer> logger) : BackgroundService
+{
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        logger.LogInformation("Starting processor...");
+
+        client.ProcessEventAsync += arg =>
+        {
+            logger.LogInformation(arg.Data.EventBody.ToString());
+            return Task.CompletedTask;
+        };
```

Comment:
When playing with this app, it seems that every time it starts up, it replays all the events that were already produced from previous runs. Is that intentional?

Reply:
Seems like a bug -- it should be checkpointing. Good catch; I may have broken something recently.

Reply:
Oh, it was never checkpointing -- there's no call to UpdateCheckpointAsync. Normally this is called after a success processing of the batch. I'm going to add this line with an explanatory comment. Good catch nonetheless.